src/types.ts

```diff
@@ -1,10 +1,12 @@
 export type MockCondition = (params: unknown[]) => boolean;
+export type MockExecution = (params: unknown[]) => Promise<unknown>;
 
 export type MockOptions = {
   persistent?: boolean;
   shouldThrow?: boolean;
   timeout?: number;
   condition?: MockCondition;
+  executeFn?: MockExecution;
```

Comment:
Related to https://github.com/VGLoic/eth-testing/pull/44#issuecomment-1173110582 :)